### PR TITLE
Read loadcombinations

### PR DESCRIPTION
**Pull request by FredericBourgeon**
Second try at fixing issue 247 with a clean recent branch based off Master.

 ### NOTE: Depends on 

 ### Issues addressed by this PR

 Closes #

 
This PR fixes the basis of issue #247. It gets the ball rolling but could be improved and build on improvements to loadcase reading.
Ignore first PR which is based off a stale version of Master.

 ### Test files

Test files (GH & rtd 2020): https://we.tl/t-htD2fmWZZ2

 ### Changelog

 ### Additional comments

### CODE REVIEW

File: Robot_Adapter/CRUD/Read/Loads/Loadcombination.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            {
+                IRobotCase rCase = rLoadCollection.Get(i) as IRobotCase;
+                if (rCase.Type == IRobotCaseType.I_CT_COMBINATION) // No support for code combinations for now as it needs building support for components
+                {
+                    IRobotCaseCombination rCaseCombination = rCase as IRobotCaseCombination;
+                    List<System.Tuple<double, ICase>> bCaseFactors = new List<System.Tuple<double, ICase>>();
+
+                    for (int j = 1; j <= rCaseCombination.CaseFactors.Count; j++)
+                    {
+                        IRobotCase rCaseIn = rLoadCollection.Get(j) as IRobotCase;
```
Comment (rwemay) [inline Robot_Adapter/CRUD/Read/Loads/Loadcombination.cs]: I don't think this returns the correct case corresponding to the case factor - see below suggested change. I ran a quick test on this and seemed to work - see before and after screenshots in the test script. 

`IRobotCase rCaseIn = rLoadCollection.Get(rCaseCombination.CaseFactors.Get(j).CaseNumber) as IRobotCase;`

image

image

File: Robot_Adapter/Robot_Adapter.csproj [hunk 2]
```diff
@@ -150,6 +150,7 @@
     <Compile Include="CRUD\Delete\_Delete.cs" />
     <Compile Include="CRUD\Create\_Create.cs" />
     <Compile Include="AdapterActions\Execute.cs" />
+    <Compile Include="CRUD\Read\Loads\Loadcombination.cs" />
```
[the remark this answers is not in the document]
Comment (FredericBourgeon) [reply Robot_Adapter/Robot_Adapter.csproj]: Sure, consistency is important. This has been addressed as per your suggestion.

File: Robot_Adapter/CRUD/Read/Loads/Loadcombination.cs (hunk 1 above)
Comment (FredericBourgeon) [reply Robot_Adapter/CRUD/Read/Loads/Loadcombination.cs]: That is correct, Robot expose loads in a load combo in the order they were defined, which can be different from their number. Your suggestion fixes this issue. Thanks for the code suggestion!

**Comment by rwemay** [comment]
Thanks @FredericBourgeon , great stuff! I also retested - works well.

**Pull request merged**